### ISSUE TITLE
Audit all non-English spellcheck dictionaries

### ISSUE THREAD

**Issue by francarotti** [rstudio/rstudio#7218]
### System details

    RStudio Edition : Desktop
    RStudio Version : 1.3.1045
    OS Version      : Windows 10 Home 2004
    R Version       : 4.0.0

### Steps to reproduce the problem

 1.  Set Brazilian Portuguese as main language for spell checking, after default installation and restarting.

 2. Create a new Rmd file with the words 
```
nao
pao
idolo
```

### Describe the problem in detail

These are some examples with words that should be accentuated. When running the spell check (F7), only the suggestions for `pao` are correct. None of the suggestions for `nao` and `idolo` are reasonable.

### Describe the behavior you expected

These are the suggestions that should appear, since only accentuation was missing.

```
não
pão
ídolo
``` 

### Additional notes

This issue was partially discussed in #7018 . I was able to make the spell checker work as intended by copying the .dic and .aff files from my LibreOffice (version 6.4) and replacing their respective in %appdata%\rstudio\dictionaries\languages-system . 

I used common words like `não` as example to illustrate that this is a problem with some words that are frequently used. 

-------------

- `[x]` I have read the guide to submitting good bug reports at https://github.com/rstudio/rstudio/wiki/Writing-Good-Bug-Reports . 
- `[x]` I have installed the latest version of RStudio and confirmed that the issue still persists.
- `[ ]` If I am reporting a RStudio crash, I have included a diagnostics report. https://support.rstudio.com/hc/en-us/articles/200321257-Running-a-Diagnostics-Report
- `[x]` I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.

**Comment by ronblum** [comment]
@francarotti Thank you for raising the issue!

@adamconroy Could this be related to #6999 or #6317?

**Comment by adamconroy** [comment]
Through the course of that thread Francesco did say that he updated the dictionaries so I believe he has the latest UTF-8 enabled ones. An issue we have is that many of the dictionaries we ship with could use some upgrading. This issue is actually fairly related to https://github.com/rstudio/rstudio/issues/6116 which will also be partially solved by just finding better dictionaries. The bonus there is that finding the writing dictionary will enabled realtime spellchecking for those languages as well. I'll get some spellcheck love in for 1.4 so we can keep improving this.

**Comment by francarotti** [comment]
Thanks for working on this issue @adamconroy! If you want someone for testing, feel free to message me.

**Comment by jmcphers** [comment]
via @adamconroy: we should look at all of our non-English dictionaries and see if we can upgrade any.

**Comment by kevinushey** [comment]
First attempt: I tried changing the main dictionary language. I didn't see Brazilian Portuguese in the list, so I clicked 'Install More Languages...'. The languages were downloaded, and then I was able to select `Portuguese (Brazil)`. I then clicked OK to accept and dismiss the options dialog.

However, after applying changes, I then return to the Global Options dialog and go to Spelling. The newly-downloaded dictionaries are not available, and the dictionary has reverted to English (Australia).

I cancel the dialog, then quit and restart RStudio. Upon restart, the newly-installed languages are then properly reflected.

I then open an R Markdown document with the contents:

`````
---
title: "Untitled"
output: html_document
---

nao
pao
idolo
`````

and try to activate spellcheck via F7. The suggested replacement appears to be mis-encoded:

<img width="426" alt="Screen Shot 2020-10-21 at 11 10 13 AM" src="https://user-images.githubusercontent.com/1976582/96762018-21e6f500-138e-11eb-87d9-60cf21728d93.png">

<img width="427" alt="Screen Shot 2020-10-21 at 11 11 26 AM" src="https://user-images.githubusercontent.com/1976582/96762211-2dd2b700-138e-11eb-8a02-56da5fcc1e4d.png">
